Plugins/FormatSwift/Plugin.swift

```diff
@@ -166,3 +167,92 @@ enum CommandError: Error {
   case lintFailure
   case unknownError(exitCode: Int32)
 }
+
+// MARK: - SwiftVersion
+
+struct SwiftVersion: Comparable {
+  var major: Int
+  var minor: Int
+
+  static func ==(_ lhs: SwiftVersion, _ rhs: SwiftVersion) -> Bool {
+    lhs.major == rhs.major
+      && lhs.minor == rhs.minor
+  }
+
+  static func <(_ lhs: SwiftVersion, _ rhs: SwiftVersion) -> Bool {
+    if lhs.major == rhs.major {
+      return lhs.minor < rhs.minor
+    } else {
+      return lhs.major < rhs.major
+    }
+  }
+}
+
+extension Package {
+
+  // MARK: Internal
+
+  /// The minimum Swift version supported by this package
+  var minimumSwiftVersion: SwiftVersion {
+    supportedSwiftVersions.sorted().first!
+  }
+
+  // MARK: Private
+
+  /// Swift versions supported by this package. Guaranteed to be non-empty.
+  ///  - This includes the `swift-tools-version` from the `Package.swift`,
+  ///    plus the Swift version of any additional version-specific Package manifest
+  ///    (e.g. `Package@swift-5.6.swift`).
+  private var supportedSwiftVersions: [SwiftVersion] {
+    guard let projectDirectory = URL(string: directory.string) else { return [] }
+
+    var supportedSwiftVersions = [
+      SwiftVersion(major: toolsVersion.major, minor: toolsVersion.minor),
+    ]
+
+    // Look for all of the package manifest files in the directory root
+    let enumerator = FileManager.default.enumerator(
+      at: projectDirectory,
+      includingPropertiesForKeys: nil,
+      options: [.skipsSubdirectoryDescendants, .skipsHiddenFiles])
+
+    while let fileURL = enumerator?.nextObject() as? URL {
+      let fileName = fileURL.lastPathComponent
+
+      guard fileName.hasPrefix("Package"), fileName.hasSuffix(".swift") else { continue }
+
+      // Parse the Swift version from the file name if it has a format like `Package@swift-5.8.swift`
+      if
+        fileName.hasPrefix("Package@swift-"),
+        let swiftVersion = parseSwiftVersion(from: fileName.dropFirst("Package@swift-".count))
+      {
+        supportedSwiftVersions.append(swiftVersion)
+      }
+
+      // Parse the Swift tools version from the file body if it starts with a comment like `// swift-tools-version: 5.8`
+      if
+        let fileContents = try? String(contentsOf: fileURL),
+        fileContents.hasPrefix("// swift-tools-version: "),
+        let swiftVersion = parseSwiftVersion(from: fileContents.dropFirst("// swift-tools-version: ".count))
+      {
+        supportedSwiftVersions.append(swiftVersion)
+      }
+    }
+
+    return supportedSwiftVersions
+  }
+
+  /// Parses a Swift version from a string like "5.8"
+  private func parseSwiftVersion(from string: Substring) -> SwiftVersion? {
+    var string = string
+
+    guard
+      string.count >= 3,
+      let major = string.popFirst().flatMap({ Int("\($0)") }),
```

Comment:
Perhaps this would be a good style guide rule 😄  (would be easy to autoformat)

Reply:
> Perhaps this would be a good style guide rule 😄 (would be easy to autoformat)

This would be cool, but do you think we will have the type information we need to know if a `String` initializer exists?

Reply:
Good point